types/ramda/index.d.ts

```diff
@@ -400,7 +401,16 @@ export function contains<T>(a: T): (list: readonly T[]) => boolean;
  * function is applied to those same arguments. The results of each branching function
  * are passed as arguments to the converging function to produce the return value.
  */
-export function converge<TArgs extends readonly any[], TResult>(after: ((...args: readonly any[]) => TResult), fns: Array<((...args: TArgs) => any)>): (...args: TArgs) => TResult;
+// tslint:disable:max-line-length
+export function converge<TArgs extends any[], TResult, R1, R2, R3, R4, R5, R6, R7, RestFunctions extends Array<(...args: TArgs) => any>>(converging: ((...args: readonly [R1, R2, R3, R4, R5, R6, R7, ...ReturnTypesOfFns<RestFunctions>]) => TResult), branches: [((...args: TArgs) => R1), ((...args: TArgs) => R2), ((...args: TArgs) => R3), ((...args: TArgs) => R4), ((...args: TArgs) => R5), ((...args: TArgs) => R6), ((...args: TArgs) => R7), ...RestFunctions]): (...args: TArgs) => TResult;
```

Comment:
This is getting painfully long, we should probably re-enable the tslint rule and break up the line, for easier understanding and maintainability.

Reply:
Multiple lines style is ugly too, IMO.
Do you have any idea how to make multi lines style less ugly and more readable.

Reply:
IMHO this is improves readability and maintainability a lot:

```ts
export function converge<
  TArgs extends any[],
  TResult,
  R1, R2, R3, R4, R5, R6, R7,
  RestFunctions extends Array<(...args: TArgs) => any>
> (
    converging: (...args: readonly [
        R1, R2, R3, R4, R5, R6, R7,
        ...ReturnTypesOfFns<RestFunctions>
    ]) => TResult,
    branches: [
        ((...args: TArgs) => R1),
        ((...args: TArgs) => R2),
        ((...args: TArgs) => R3),
        ((...args: TArgs) => R4),
        ((...args: TArgs) => R5),
        ((...args: TArgs) => R6),
        ((...args: TArgs) => R7),
        ...RestFunctions
    ]
): (...args: TArgs) => TResult;
```

I agree that it's ugly, but it's not going to be pretty any way.

Alternatively for converging's args we could do:

```ts
    converging: (
        ...args: readonly [
            R1, R2, R3, R4, R5, R6, R7,
            ...ReturnTypesOfFns<RestFunctions>
        ]
    ) => TResult,
```

I don't think it adds anymore readability but I guess some people would call it "more correct". I like the leaner, first example

Reply:
ok, I will modify the functions which length violate tslint.